cylc/flow/scripts/lint.py

```diff
@@ -166,6 +167,122 @@
         )
     },
 }
+RULESETS = ['728', 'style', 'all']
+EXTRA_TOML_VALIDATION = {
+    'ignore': {
+        lambda x: re.match(r'[A-Z]\d\d\d', x):
+            '{item} not valid: Ignore codes should be in the form X001',
+        lambda x: x in [
+            f'{i["purpose"]}{i["index"]:03d}'
+            for i in parse_checks(['728', 'style']).values()
+        ]:
+            '{item} is a not a known linter code.'
+    },
+    'rulesets': {
+        lambda item: item in RULESETS:
+            '{item} not valid: Rulesets can be '
+            '\'728\', \'style\' or \'all\'.'
+    },
+    # consider checking that item is file?
+    'exceptions': {}
+}
+
+
+class tomlDataValidatationFailure(Exception):
+    ...
+
+
+def validate_toml_items(tomldata):
+    """Check that all tomldata items are lists of strings
+
+    Explicitly checks and raises if tomldata not:
+        - str in EXTRA_TOML_VALIDATION.keys()
+        - item is not a list of strings.
+
+    Plus additional validation for each set of values.
+    """
+    for key, items in tomldata.items():
+        # Key should only be one of the allowed keys:
+        if key not in EXTRA_TOML_VALIDATION.keys():
+            raise tomlDataValidatationFailure(
+                f'Only {[*EXTRA_TOML_VALIDATION.keys()]} '
+                f'allowed as toml sections but you used {key}'
+            )
```

Comment:
Any reason these can't be the built-in exception classes? E.g. `ValueError`

Reply:
No particular reason. When I started it I wanted a unique name, and possibly to consider some extra logic, but I didn't use it in the end. Think it should go?

Reply:
Might as well remove it to tidy up